sweepbatcher/sweep_batch.go

```diff
@@ -1900,7 +1903,11 @@ func (b *batch) handleSpend(ctx context.Context, spendTx *wire.MsgTx) error {
 		notifyList = make([]sweep, 0, len(b.sweeps))
 	)
 	b.batchTxid = &txHash
-	b.batchPkScript = spendTx.TxOut[0].PkScript
+	if len(spendTx.TxOut) > 0 {
+		b.batchPkScript = spendTx.TxOut[0].PkScript
+	} else {
+		b.log.Warnf("transaction %v has no outputs", txHash)
```

Comment:
Instead of warning I think we'd be better of erroring out here to not intentionally allow burns.

Reply:
This is the spending handler. Since the transaction is already in the mempool, there's nothing we can do at this point. The only way this code could be reached is if the counterparty executed the timeout path and chose to burn the funds, which we cannot prevent anyway.

Reply:
Thank you for the explanation!